File: 118.pascals-triangle.py
#
# @lc app=leetcode id=118 lang=python3
#
# [118] Pascal's Triangle
#

# @lc code=start
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        elif numRows == 1:
            return [[1]]
        elif numRows == 2:
            return [[1], [1, 1]]
        rows = [[1], [1, 1]]
        for i in range(2, numRows):
            rows.append([1] * (len(rows[i-1])+1))
            k = 1
            for j in range(len(rows[i-1])-1):
                rows[i][k] = rows[i-1][j] + rows[i-1][j+1]
                k = k + 1
        return rows


            
# @lc code=end

File: code/118.pascals-triangle.cpp
/*
 * @lc app=leetcode id=118 lang=cpp
 *
 * [118] Pascal's Triangle
 */
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows == 0) return {};
        vector<vector<int>> result = {{1}};
        for(int i = 1;i < numRows;++i) {
            vector<int> row;
            row.push_back(1);
            for(int j = 0;j < result[i - 1].size() - 1;++j) {
                row.push_back(result[i - 1][j] + result[i - 1][j + 1]);
            }
            row.push_back(1);
            result.push_back(row);
        }
        
        return result;
    }
};

